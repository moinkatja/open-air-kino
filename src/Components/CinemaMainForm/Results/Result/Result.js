import React from 'react';
import '../CinemaProfile/CinemaProfile';
import classes from "./Result.module.css";

function Result(props) {

    return (
        <div
            className={props.id === props.selectedCinema ? [classes.ResultButton, classes.ResultButton_Active].join(" ") : [classes.ResultButton, classes.ResultButton_NotActive].join(" ")}
            id={props.id}
            onClick={props.clickedResult} >
                
            <img
                className={props.id === props.selectedCinema ? classes.KinoThumbnail : [classes.KinoThumbnail, classes.KinoThumbnail_NotActive].join(" ")}
                src={props.pic} alt="Kino Thumbnail" />
            <h4><strong>{props.name}</strong></h4>
            <p> {props.city}, {props.street} </p>
            
            <button className={props.likeBtn === "Like" ? classes.LikeButton : [classes.LikeButton, classes.LikeButton_NotActive].join(" ")} key={props.id}
                onClick={props.clickedLike}
            > {props.likeBtn}</button>

        </div>
    )
}

export default Result
