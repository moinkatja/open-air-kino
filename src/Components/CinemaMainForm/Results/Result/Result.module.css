.ResultButton {
    padding: 5px;
    width: 90%;
    height: 110px;
    margin: 5px auto;
    opacity: 0.95;
    border-radius: 10px;
    text-align: center;
    transition: 0.3s ease-in-out;
    max-width: 500px;
}

.ResultButton_NotActive {
    border: 1px dotted teal;
    background-color: #80dfd7;
    color: rgb(56, 58, 58);
    cursor: pointer;
}

.ResultButton_Active {
    border: 1px solid #70e7dd;
    background-color: teal;
    color: white;
    cursor: not-allowed;
}

.ResultButton_NotActive:hover {
    background-color: rgb(131, 109, 168);
    color: white;
}

.KinoThumbnail {
    border-radius: 50%;
    width: 80px;
    height: 80px;
    float: left;
    margin-top: 10px;
    margin-left: 2rem;
    opacity: 0.8;
}

.KinoThumbnail_NotActive {
    filter: grayscale(1);
}

.LikeButton {
    float: right;
    margin-top: -3rem;
    width: 50px;
    background-color: white;
    color: green;
    border: none;
    border-radius: 5px;
    cursor: pointer;
    margin-right: .3rem;
}

.LikeButton_NotActive {
    background-color: rgb(88, 62, 99);
    color: white;
}

@media (max-width: 1200px) {
    .ResultButton {
        height: 100px;
    }

    .ResultButton p{
        display: none;
    } 

    .LikeButton {
        float: none;
    }

    .KinoThumbnail {
        margin-left: 1rem;
    }
}

@media (max-width: 1000px) {
    .ResultButton p {
        padding: 5px;
        display: block;
        font-size: .8rem;
    } 

    .KinoThumbnail {
        margin-left: 2rem;
        margin-top: 5px;
    }

    .LikeButton {
        float: right;
    }
}

@media (max-width: 420px) {
    .ResultButton p {
        display: none;
    } 

    .LikeButton {
        float: none;
    }
}
  