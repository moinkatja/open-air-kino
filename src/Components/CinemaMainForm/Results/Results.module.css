.Results {
    background-color: white;
    border-radius: 10px;
    margin-left: 1rem;
    height: 100%;
} 

a {
    text-decoration: none;
}

a:visited {
    color: rgb(63, 60, 60);
}

.NoFavMessage {
    align-items: center;
    margin-top: 10%;
}


@media (max-width: 1000px) {

    .Results {
      width: 100%;
      padding-top: 1rem;
      padding-bottom: .5rem;
      margin-left: 0;
      border-radius: 0;
      margin-top: 1rem;  
    }
}
