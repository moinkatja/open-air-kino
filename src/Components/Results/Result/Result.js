import React from 'react';
import '../../CinemaProfile/CinemaProfile';
import { BrowserRouter, NavLink } from "react-router-dom";

function Result(props) {
    return (
        <BrowserRouter>
        <NavLink to={
            { pathname: `/cinemas/${props.id}` }}>
            <div
                className={props.id === props.activeCinema ? "ActiveResultButton" : "ResultButton"}
                id={props.id}
                onClick={props.clicked} >
                <img
                    className={props.id === props.activeCinema ? "ActiveKinoThumbnail" : "KinoThumbnail"}
                    src={props.pic} alt="Kino Thumbnail" />
                <h4><strong>{props.name}</strong></h4> <p> {props.city}, {props.street} </p>

                <button className={props.liked === "Like" ? "ActiveLikeButton" : "NotActiveLikeButton"} key={props.id}
                    onClick={props.favorites}
                >  {props.liked}</button>
            </div>
        </NavLink>
    </BrowserRouter>
    )
}

export default Result
