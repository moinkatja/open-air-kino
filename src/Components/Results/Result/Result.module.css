.ResultItem {
    padding: 5px;
    border: 1px dotted grey;
    display: flex;
    width: 80%;
    margin: 5px auto;
    justify-content: center;
    background-color: #70e7dd;
    opacity: 0.9;
    border-radius: 10px;
    cursor: pointer;
}



.KinoThumbnail {
    border-radius: 50%;
    width: 80px;
    height: 80px;
    margin-right: 2rem;
    align-self: center;
}

h2 {
    font-size: 1.2rem;
}

h3 {
    font-size: .8rem;
}