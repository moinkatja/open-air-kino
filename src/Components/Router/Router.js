import React from 'react';
import { BrowserRouter, Switch, Route } from "react-router-dom";

import App from "../../App";
import CinemaProfile from "../CinemaProfile/CinemaProfile";

const Router = () => (
  
  <BrowserRouter>
    <Switch>
      <Route path="/" component={App} exact />
      <Route path="id"  component={CinemaProfile} />
    </Switch>
  </BrowserRouter>
);

export default Router;
