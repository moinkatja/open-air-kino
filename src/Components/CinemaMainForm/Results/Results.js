import React, { Component } from 'react';
import Result from "./Result/Result";
import Pagination from "./Pagination/Pagination";
import CinemaProfile from "./CinemaProfile/CinemaProfile";
import WelcomePage from "./WelcomePage/WelcomePage";
import { NavLink } from "react-router-dom";
import { getFavs, getCinemaRegion, getCinemaDetails, getResultsPerPage, calculateCurrentPage } from "../../../services";
import ControlButtons from "./ControlButtons/ControlButtons"

import classes from "./Results.module.css";

class Results extends Component {
    constructor(props) {
        super(props);

        this.state = {
            favorites: [],
            displayedCinemas: [],
            selectedCinema: null,
            currentPage: null,
            resultsPerPage: 5,
        };
    }

    componentDidMount() {
        const getFavs = JSON.parse(localStorage.getItem("favorites")) || "[]";

        this.setState({
            favorites: getFavs,
            displayedCinemas: this.props.cinemas,
        });


        this.getPagination();
        window.addEventListener("resize", this.getPagination.bind(this));

        (this.props.tab === "cinemas" || this.props.tab === "favorites") && (getCinemaDetails(this.props.cinemas, this.props.cinemaId)) ?
            this.setState({
                selectedCinema: this.props.cinemaId,
                currentPage: calculateCurrentPage(this.props.cinemaId, this.state.resultsPerPage)
            }) :
            this.setState({
                selectedCinema: null,
                currentPage: 1
            });
    }

    goHome = () => {
        this.setState({
            displayedCinemas: this.props.cinemas,
            selectedCinema: null,
            currentPage: 1,
        })
    }

    //Pagination depending on the screen size

    getPagination = () => {
        this.setState({
            resultsPerPage: getResultsPerPage()
        });
    }


    //Handler for dropdown with regions

    getRegion = (e) => {
        e.preventDefault();
        const region = e.target.value;

        if (region === "") {
            this.setState({
                displayedCinemas: this.props.cinemas,
                currentPage: 1,
                selectedCinema: this.props.cinemas[0].id,
            })
        }
        else {
            this.setState({
                displayedCinemas: getCinemaRegion(this.props.cinemas, region),
                currentPage: 1,
                selectedCinema: getCinemaRegion(this.props.cinemas, region)[0].id,
            });
        }
    }

    //Change active cinema by clickng on it

    changeActiveCinema = (e) => {
        const cinemaId = e.target.id;
        this.setState({
            selectedCinema: cinemaId
        });
    }


    handlePageClick = (e) => {
        e.preventDefault();
        const pageNum = Number(e.target.id);
        this.setState({
            currentPage: pageNum
        });
    }

    //Add cinemaId to favorites state and toggle betweeb "Like" and "Dislike"

    addFavorite = (id) => {
        if (!this.state.favorites.includes(id)) {
            this.setState({
                favorites: [...this.state.favorites, id],
            })
        }

        else {
            this.setState({
                favorites: this.state.favorites.filter(function (cinema) {
                    return cinema !== id
                }),
            })
        }
    }

    //Show favorites by clicking on the fav button

    showFavorites = () => {
        this.setState({
            currentPage: 1,
            selectedCinema: getFavs(this.props.cinemas, this.state.favorites)[0].id,
        });
    }

    //Save favorites to LocalStorage

    componentDidUpdate() {
        localStorage.setItem("favorites", JSON.stringify(this.state.favorites));
    }

    componentWillUnmount() {
        window.removeEventListener("resize", this.getPagination.bind(this));
    }

    render() {

        let cinemasToDisplay;
        let cinemaDetails;


        switch (this.props.tab) {
            case "favorites": {
                cinemasToDisplay = getFavs(this.props.cinemas, this.state.favorites)
                this.state.favorites.includes(this.props.cinemaId) ?
                    cinemaDetails = getCinemaDetails(this.props.cinemas, this.props.cinemaId) :
                    cinemaDetails = null;
                break;
            }
            default: {
                cinemasToDisplay = this.state.displayedCinemas;
                this.props.cinemaId ?
                    cinemaDetails = getCinemaDetails(this.props.cinemas, this.props.cinemaId) :
                    cinemaDetails = getCinemaDetails(this.props.cinemas, this.state.selectedCinema);
            }
        }

        let indexOfLastResult = this.state.currentPage * this.state.resultsPerPage;
        let indexOfFirstResult = indexOfLastResult - this.state.resultsPerPage;
        let currentResult = cinemasToDisplay.slice(indexOfFirstResult, indexOfLastResult);

        console.log(cinemaDetails)
        console.log(this.state.selectedCinema)

        return (
            <section className={classes.CinemaApp}>
                <div className={classes.Results}>
                    <ControlButtons
                        favorites={this.state.favorites}
                        cinemas={this.props.cinemas}
                        clickedHomeBtn={this.goHome}
                        clickedFavBtn={this.showFavorites.bind(this)}
                        getRegion={this.getRegion}
                        tab={this.props.tab}
                    />

                    {(currentResult.length > 0) ?
                        currentResult.map((cinema, id) =>
                            <NavLink key={id}
                                to={`/${this.props.tab}/${cinema.id}`} >
                                <Result
                                    selectedCinema={this.state.selectedCinema}
                                    key={id}
                                    id={cinema.id}
                                    name={cinema.name}
                                    tel={cinema.tel}
                                    street={cinema.street}
                                    city={cinema.city}
                                    pic={cinema.pic}
                                    clickedResult={this.changeActiveCinema.bind(this)}
                                    clickedLike={() => this.addFavorite.bind(this)(cinema.id)}
                                    likeBtn={(this.state.favorites).includes(cinema.id) ? "Dislike" : "Like"}
                                />

                            </NavLink>
                        ) : <h4 className={classes.NoFavMessage}>Sorry, you do not have any favorites yet</h4>
                    }

                    <Pagination
                        cinemas={cinemasToDisplay}
                        clickedPage={this.handlePageClick.bind(this)}
                        currentPage={this.state.currentPage}
                        resultsPerPage={this.state.resultsPerPage}
                    />
                </div>
                <div>
                    {
                        cinemaDetails ?
                            <CinemaProfile
                                cinemaDetails={cinemaDetails}
                            /> : <WelcomePage />
                    }
                </div >
            </section>
        )
    }
}

export default Results;
