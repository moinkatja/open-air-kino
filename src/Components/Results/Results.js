import React, { Component } from 'react';
import Result from "./Result/Result";
import Pagination from "./Pagination/Pagination";
import CinemaProfile from "../CinemaProfile/CinemaProfile";
import WelcomePage from "../WelcomePage/WelcomePage";
import { NavLink } from "react-router-dom";
import { getFavs, getCinemaRegion, getCinemaDetails, calculateCurrentPage, getResultsPerPage } from "../../services";
import ControlButtons from "../ControlButtons/ControlButtons"

import classes from "./Results.module.css";

class Results extends Component {
    _isMounted = false;
    constructor(props) {
        super(props);

        this.state = {
            favorites: [],
            displayedCinemas: [],
            selectedCinema: null,
            currentPage: 1,
            resultsPerPage: null,
        };
    }

    componentDidMount() {
        this._isMounted = true;
        const getFavs = JSON.parse(localStorage.getItem("favorites")) || "[]";

        this.setState({
            favorites: getFavs,
            displayedCinemas: this.props.cinemas,
            currentPage: 1,
        });

        this.props.cinemaId ? this.setState({ selectedCinema: this.props.cinemaId }) : this.setState({ selectedCinema: null });

        this.getPagination();
        window.addEventListener("resize", this.getPagination.bind(this));
    }

    goHome = () => {
        this.setState({
            displayedCinemas: this.props.cinemas,
            selectedCinema: null,
            currentPage: 1,
        })
    }

    //Handler for dropdown with regions

    getRegion = (e) => {
        e.preventDefault();
        const region = e.target.value;

        if (region === "") {
            this.setState({
                loading: false,
                displayedCinemas: this.props.cinemas,
                currentPage: 1,
            })
        }
        else {
            this.setState({
                loading: false,
                displayedCinemas: getCinemaRegion(this.props.cinemas, region),
                currentPage: 1,
                selectedCinema: getCinemaRegion(this.props.cinemas, region)[0].id,
            });
        }
    }

    //Change active cinema by clickng on it

    changeActiveCinema = (e) => {
        const cinemaId = e.target.id;
        this.setState({
            selectedCinema: cinemaId
        });
    }

    //Pagination depending on the screen size

    getPagination = () => {
        this.setState({
            resultsPerPage: getResultsPerPage()
        });
    }

    handlePageClick = (e) => {
        e.preventDefault();
        const pageNum = Number(e.target.id);
        this.setState({
            currentPage: pageNum
        });
    }

    //Add cinemaId to favorites state and toggle betweeb "Like" and "Dislike"

    addFavorite = (id) => {
        if (!this.state.favorites.includes(id)) {
            this.setState({
                favorites: [...this.state.favorites, id]
            })
        }
        else {
            this.setState({
                favorites: this.state.favorites.filter(function (cinema) {
                    return cinema !== id
                })
            })
        }
    }

    //Show favorites by clicking on the fav button

    showFavorites = () => {
        this.setState({
            currentPage: 1,
        });
    }

    //Save favorites to LocalStorage

    componentDidUpdate() {
        localStorage.setItem("favorites", JSON.stringify(this.state.favorites));
    }

    componentWillUnmount() {
        window.removeEventListener("resize", this.getPagination.bind(this));
        this._isMounted = false;
    }

    render() {
        let cinemasToDisplay;
        let cinemaDetails;

        this.props.cinemaId ?
            cinemaDetails = getCinemaDetails(this.props.cinemas, this.props.cinemaId) :
            cinemaDetails = getCinemaDetails(this.props.cinemas, this.state.selectedCinema);

        switch (this.props.tab) {
            case "favorites": {
                cinemasToDisplay = getFavs(this.props.cinemas, this.state.favorites)
                if (this.state.favorites.includes(this.props.cinemaId) === false){ cinemaDetails = null }
                break;
            }
            default: cinemasToDisplay = this.state.displayedCinemas;
        }

       /*  let currentPage;
        this.props.cinemaId ? currentPage = calculateCurrentPage(this.props.cinemaId, this.state.resultsPerPage) :
        currentPage = this.state.currentPage; */

        let indexOfLastResult = this.state.currentPage * this.state.resultsPerPage;
        let indexOfFirstResult = indexOfLastResult - this.state.resultsPerPage;
        let currentResult = cinemasToDisplay.slice(indexOfFirstResult, indexOfLastResult);

        // : (this.props.cinemaId ? getFavs(this.state.cinemasInitial, this.props.cinemaId):  this.state.cinemas)
        return (
            <section className={classes.CinemaApp}>
                <div className={classes.Results}>
                    <ControlButtons
                        favorites={this.state.favorites}
                        cinemas={this.props.cinemas}
                        selectedCinema={cinemaDetails}
                        clickedHomeBtn={this.goHome}
                        clickedFavBtn={this.showFavorites.bind(this)}
                        getRegion={this.getRegion}
                        tab={this.props.tab}
                    />

                    {(currentResult.length > 0) ?
                        currentResult.map((cinema, id) =>
                            <NavLink key={id}
                                to={`/${this.props.tab}/${cinema.id}`} >
                                <Result
                                    selectedCinema={this.state.selectedCinema}
                                    name={cinema.name}
                                    tel={cinema.tel}
                                    street={cinema.street}
                                    city={cinema.city}
                                    id={cinema.id}
                                    key={id}
                                    pic={cinema.pic}
                                    clickedResult={this.changeActiveCinema.bind(this)}
                                    clickedLike={() => this.addFavorite.bind(this)(cinema.id)}
                                    likeBtn={(this.state.favorites).includes(cinema.id) ? "Dislike" : "Like"}
                                />

                            </NavLink>
                        ) : <h4 className={classes.NoFavMessage}>Sorry, you do not have any favorites yet</h4>
                    }

                    <Pagination
                        cinemas={cinemasToDisplay}
                        clickedPage={this.handlePageClick.bind(this)}
                        currentPage={this.state.currentPage}
                        resultsPerPage={this.state.resultsPerPage}
                    />
                </div>
                <div>
                    {
                        cinemaDetails ?
                            <CinemaProfile
                                cinemaDetails={cinemaDetails}
                            /> : <WelcomePage />
                    }
                </div >
            </section>
        )
    }
}

export default Results;
