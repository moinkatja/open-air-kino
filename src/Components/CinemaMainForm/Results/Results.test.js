import React from 'react';
import Results from "./Results";
import { render, unmountComponentAtNode } from "react-dom";
import renderer from 'react-test-renderer';
import Router from "../../Router/Router";
import MainForm from "../CinemaMainForm";
import Result from "./Result/Result";

let container = null;
beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
});

afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
});

test('Results render correctly', () => {
    const loading = false;
    const favorites = [1];
    const cinemas = [
        { "id": "0", "name": "Open-Air-Schanzenkino", "region": "Hamburg", "postcode": 20357, "city": "Hamburg", "street": "Sternschanze 1A", "pic": "https://i.postimg.cc/zvcsTqLQ/kino2.jpg" },
        { "id": "1", "name": "Open Air Kino in Dornumersiel", "region": "Niedersachsen", "postcode": 26553, "city": "Dornumersiel", "street": "Seeparkbühne", "pic": "https://i.postimg.cc/MTWJSP29/kino1.jpg" },
        { "id": "2", "name": "Biesdorfer Parkbühne", "region": "Berlin", "postcode": 12683, "city": "Berlin", "street": "Nordpromenade 5", "pic": "https://i.postimg.cc/j22VWDjN/kino3.jpg" },
    ]
    const tree = renderer.create(
        <Router>
            <MainForm loading={loading}>
            </MainForm>
        </Router>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});