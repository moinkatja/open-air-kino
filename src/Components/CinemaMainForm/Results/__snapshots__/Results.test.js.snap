// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results render correctly 1`] = `
Array [
  <div
    className="Modal"
    style={
      Object {
        "opacity": "0",
        "transform": "translateY(-100vh)",
      }
    }
  />,
  <section
    className="CinemaApp"
  >
    <header
      className="Header"
    >
      <img
        alt="Logo"
        className="Logo"
        src="logo.png"
      />
      <h1
        className="Title"
      >
        OpenAir Kino 
      </h1>
      <a
        className="GitHubLink"
        href="https://github.com/moinkatja/open-air-kino"
        title="GitHub Project"
      >
        <svg
          dangerouslySetInnerHTML={
            Object {
              "__html": "<path fill-rule=\\"evenodd\\" clip-rule=\\"evenodd\\" fill=\\"darkgrey\\" d=\\"M25 10c-8.3 0-15 6.7-15 15c0 6.6 4.3 12.2 10.3 14.2c.8.1 1-.3 1-.7v-2.6c-4.2.9-5.1-2-5.1-2c-.7-1.7-1.7-2.2-1.7-2.2c-1.4-.9.1-.9.1-.9c1.5.1 2.3 1.5 2.3 1.5c1.3 2.3 3.5 1.6 4.4 1.2c.1-1 .5-1.6 1-2c-3.3-.4-6.8-1.7-6.8-7.4c0-1.6.6-3 1.5-4c-.2-.4-.7-1.9.1-4c0 0 1.3-.4 4.1 1.5c1.2-.3 2.5-.5 3.8-.5c1.3 0 2.6.2 3.8.5c2.9-1.9 4.1-1.5 4.1-1.5c.8 2.1.3 3.6.1 4c1 1 1.5 2.4 1.5 4c0 5.8-3.5 7-6.8 7.4c.5.5 1 1.4 1 2.8v4.1c0 .4.3.9 1 .7c6-2 10.2-7.6 10.2-14.2C40 16.7 33.3 10 25 10z\\"/>",
            }
          }
          focusable={false}
          height="50"
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "transform": "rotate(360deg)",
            }
          }
          viewBox="0 0 50 50"
          width="50"
          xmlns="http://www.w3.org/2000/svg"
        />
      </a>
    </header>
    <div
      className="Loader"
    >
      Loading...
    </div>
    <footer
      className="Footer"
    >
      <p>
        <sup>
          *
        </sup>
         It is just a demo app. All the data is provided "as is", without any conditions, warranties or other terms of any kind. 
      </p>
    </footer>
  </section>,
]
`;
